module/persistent.lua

```diff
@@ -0,0 +1,192 @@
+        client.maximized = client_get_xproperty(client, "maximized", "boolean")
+        client.maximized_horizontal = client_get_xproperty(client, "maximized_horizontal", "boolean")
+        client.maximized_vertical = client_get_xproperty(client, "maximized_vertical", "boolean")
+        client.sticky = client_get_xproperty(client, "sticky", "boolean")
+        client.floating = client_get_xproperty(client, "floating", "boolean")
+
+        client.x = client_get_xproperty(client, "x", "number")
+        client.y = client_get_xproperty(client, "y", "number")
+        client.width = client_get_xproperty(client, "width", "number")
+        client.height = client_get_xproperty(client, "height", "number")
+
+        local parent = client
+        local bling_tabbed_clients_amount = client_get_xproperty(parent, "bling_tabbed_clients_amount", "number") or 0
+        for i = 1, bling_tabbed_clients_amount, 1 do
+            local child_window = tonumber(client_get_xproperty(parent, "bling_tabbed_client_" .. i, "string"))
+            for index, child in ipairs(capi.client.get()) do
+                if child.window == child_window then
+                    local tab_index = client_get_xproperty(client, "bling_tabbed_focused_idx", "number")
+                    if not parent.bling_tabbed and not child.bling_tabbed then
+                        tabbed.init(parent)
+                        tabbed.add(child, parent.bling_tabbed)
+                        gears.timer.delayed_call(function()
+                            tabbed.switch_to(parent.bling_tabbed, tab_index)
+                        end)
+                    end
+                    if not parent.bling_tabbed and child.bling_tabbed then
+                        tabbed.add(parent, child.bling_tabbed)
+                        gears.timer.delayed_call(function()
+                            tabbed.switch_to(child.bling_tabbed, tab_index)
+                        end)
+                    end
+                    if parent.bling_tabbed and not child.bling_tabbed then
+                        tabbed.add(child, parent.bling_tabbed)
+                        gears.timer.delayed_call(function()
+                            tabbed.switch_to(parent.bling_tabbed, tab_index)
+                        end)
+                    end
+                   child:tags({})
+                end
+            end
+        end
+
+        gears.timer.delayed_call(function()
+            local tags_count = client_get_xproperty(client, "tags_count", "number") or 0
+            local tags = {}
+            for i = 1, tags_count, 1 do
+                local tag_index = client_get_xproperty(client, "tag_" .. i, "number")
+                table.insert(tags, capi.screen[client.screen].tags[tag_index])
+            end
+
+            client.first_tag = tags[1]
+            client:tags(tags)
+        end)
+    end
+end
+
+local function new(args)
+    local ret = gears.object{}
```

Comment:
What's the point of this object?

It seems the module is only composed of private helper functions.
Why do you need an instance (singleton pattern, I guess (?)) to manage that? From what I can see, `instance` doesn't hold any value nor any exposed methods. ~~It's basically an empty object.~~ Nope, actually it has its own `enable` method (that returns the object instance itself).

Reply:
> It seems the module is only composed of private helper functions.

Yeah, right. I initially wanted to have the user call :restore() on his own, but forgot to remove that part. 

> singleton pattern, I guess (?)) 
Yeah.